### PR TITLE
Update deps to aws v6

### PR DESCRIPTION
**Pull request by mnlumi**
### Related issues (optional)

Resolves https://github.com/pulumi/pulumi-eks/pull/901

### CODE REVIEW

**Review by danielrbradley — APPROVED** [review]

**Review by iwahbe — APPROVED** [review]
LGTM

**Comment by t0yv0** [comment]
Looking at failures in https://github.com/pulumi/pulumi-aws/issues/2774

File: provider/go.mod [hunk 1]
```diff
@@ -1,9 +1,211 @@
 module github.com/pulumi/pulumi-eks/provider
 
-go 1.16
+go 1.20
 
 require (
 	github.com/pkg/errors v0.9.1
-	github.com/pulumi/pulumi/pkg/v3 v3.64.0
-	github.com/pulumi/pulumi/sdk/v3 v3.64.0
+	github.com/pulumi/pulumi/pkg/v3 v3.81.0
```
Comment (t0yv0) [inline provider/go.mod:7]: I see this update is part of this PR, awesome. 

Are we merging this soon-ish? Note also that master branch stopped compiling.

Comment (thomas11) [reply provider/go.mod:7]: Yes. For some definition of soon-ish :)

**Comment by thomas11** [comment]
Superceded by #910

**Pull request closed without merging**